Create ui-c8y-1019-12-0-services-registry-service-that-introduces-a-hook-do-register-services-in.md (#1823)
SoftwareAG/c8y-docs: content/change-logs/application-enablement/ui-c8y-1019-12-0-services-registry-service-that-introduces-a-hook-do-register-services-in.md

---
date: ""

> title: Introduce service registry with a hook for registering services

Reviewer: I would just say "New service registry". Additional details will be provided in the description.

Reviewer 2: Well when you just say new service registry it is too vague and the idea cannot be understood. It is a registry only dev people will need and use so the details can be a little more technical and saying that it is providing a hook gives the reader the idea that it can register services via hook, otherwise it is not clear what this registry is doing.

Reviewer 3: I also vote for providing more details in the title.

Reviewer: OK Fine for me

> title: Introduce services registry with a hook to register services in the core platform

Reviewer 3: ```suggestion
title: Introduce service registry with a hook for registering services
```

> title: Introduce services registry service with hook to register services in core platform

Reviewer 2: ```suggestion
title: Introduce services registry with a hook to register services in the core platform
```

> title: Services registry service that introduces a hook do register services in the core platform. [GRAFT][release/cd] (#5869)

Author: The Claude AI suggested to use the following title:
```suggestion
title: Introduce services registry service with hook to register services in core platform
```

product_area: Application enablement & solutions
change_type:
  - value: change-QHu1GdukP
    label: Feature
component:
  - value: component-YbYJ3gLU_
    label: Web SDK
build_artifact:
  - value: tc-pjJiURv9Y
    label: ui-c8y
ticket: DM-3092
version: 1019.12.0
---

> Services registry service that introduces a hook do register services in the core platform. [GRAFT][release/cd] (#5869)

Reviewer 2: ```suggestion
To enable better management and discoverability of services in the core platform, a new services registry has been introduced. This service provides a hook that allows services to register themselves with the core platform.
```

> Services registry service that introduces a hook do register services in the core platform. [GRAFT][release/cd] (#5869)

Author: The Claude AI suggested to use the following body:
```suggestion
To enable better management and discoverability of services in the core platform, a new services registry service has been introduced. This service provides a hook that allows services to register themselves with the core platform. With this change, services can now be easily registered and discovered within the system, improving overall service management and integration. This impacts the core platform and any services that need to be registered, requiring them to utilize the new registration hook provided by the services registry service.
```

> To improve the management and discoverability of services in the Cumulocity IoT platform, a new service registry has been introduced. This registry provides a hook that allows services to register themselves with the platform.

Reviewer: To improve the management and discoverability of services in Cumulocity IoT, a new service registry has been introduced that provides a hook allowing services to register with the platform.  -> do we need the detail of the hook? That just makes it more complex to read and if ppl are interested to learn about the implementation they can look it up in the doc.  I would prefer to just say "...registry has been introduced allowing services to register with the platform" for better readability

Reviewer: that would also be an option: "To improve the management and discoverability of services in Cumulocity IoT, a new service registry has been introduced allowing services to register with the platform via hooks."

Reviewer 2: Upper comment, we do because otherwise registering a service where, for what reason, it is not getting clear.

Reviewer: You're right. Should we go for the proposal in my 2nd comment or with the initial draft?

Reviewer 2: The rewrite is fine, just removing the second sentence. I have changed that

> To improve the management and discoverability of services in the Cumulocity IoT platform, a new service registry has been introduced. This registry provides a hook that allows services to register themselves with the platform.

Reviewer 2: ```suggestion
To improve the management and discoverability of services in Cumulocity IoT, a new service registry has been introduced allowing services to register with the platform via hooks.
```

> To enable better management and discoverability of services in the core platform, a new services registry has been introduced. This service provides a hook that allows services to register themselves with the core platform.

Reviewer 3: ```suggestion
To improve the management and discoverability of services in the Cumulocity IoT platform, a new service registry has been introduced. This registry provides a hook that allows services to register themselves with the platform.
```
